ReactAndroid/src/main/java/com/facebook/react/modules/clipboard/ClipboardModule.java

```diff
@@ -43,28 +44,42 @@ public String getName() {
   }
 
   private ClipboardManager getClipboardService() {
-    ReactApplicationContext reactContext = getReactApplicationContext(); 
+    ReactApplicationContext reactContext = getReactApplicationContext();
     return (ClipboardManager) reactContext.getSystemService(reactContext.CLIPBOARD_SERVICE);
   }
 
+  private String getStringInternal(){
+    ClipboardManager clipboard = getClipboardService();
+    ClipData clipData = clipboard.getPrimaryClip();
+    if (clipData == null) {
+      return "";
+    }
+    if (clipData.getItemCount() >= 1) {
+      ClipData.Item firstItem = clipboard.getPrimaryClip().getItemAt(0);
+      return "" + firstItem.getText();
+    } else {
+      return "";
+    }
+  }
+
   @ReactMethod
   public void getString(Callback cb) {
     try {
-      ClipboardManager clipboard = getClipboardService();
-      ClipData clipData = clipboard.getPrimaryClip();
-      if (clipData == null) {
-        cb.invoke("");
-        return;
-      }
-      if (clipData.getItemCount() >= 1) {
-        ClipData.Item firstItem = clipboard.getPrimaryClip().getItemAt(0);
-        String text = "" + firstItem.getText();
-        cb.invoke(text);
-      } else {
-        cb.invoke("");
-      }
+      String ret = this.getStringInternal();
+      cb.invoke(ret);
+    } catch(Exception e) {
+      FLog.w(ReactConstants.TAG, "Cannot get clipboard contents: " + e.getMessage());
+    }
+  }
+
+  @ReactMethod
+  public void getStringSync(Promise promise){
+    try {
+      String ret = this.getStringInternal();
+      promise.resolve(ret);
     } catch(Exception e) {
       FLog.w(ReactConstants.TAG, "Cannot get clipboard contents: " + e.getMessage());
+      promise.reject(e.getMessage());
```

Comment:
Directly pass the exception to `Promise.reject`, e.g. -  no need for `e.getMessage()`